tests/unit/common/test_dynamo_handler.py

```diff
@@ -0,0 +1,35 @@
+import os
+import unittest
+
+from botocore.stub import Stubber
+
+from matrix.common.dynamo_handler import DynamoHandler
+from matrix.common.dynamo_handler import StateTableField
+
+
+class TestDynamoHandler(unittest.TestCase):
+    """
+    Environment variables are set in tests/__init__.py
+    """
+    def setUp(self):
+        self.handler = DynamoHandler()
+
+    def test_init_state_table(self):
+        mock_client = Stubber(self.handler._state_table.meta.client)
+        request_id = "test_id"
+        num_bundles = 2
+        expected_params = {
+            'TableName': os.environ['DYNAMO_STATE_TABLE_NAME'],
+            'Item': {
+                StateTableField.REQUEST_ID.value: request_id,
+                StateTableField.EXPECTED_WORKER_EXECUTIONS.value: 0,
+                StateTableField.COMPLETED_WORKER_EXECUTIONS.value: 0,
+                StateTableField.EXPECTED_MAPPER_EXECUTIONS.value: num_bundles,
+                StateTableField.COMPLETED_MAPPER_EXECUTIONS.value: 0,
+                StateTableField.EXPECTED_REDUCER_EXECUTIONS.value: 1,
+                StateTableField.COMPLETED_REDUCER_EXECUTIONS.value: 0
+            }
+        }
+        mock_client.add_response("put_item", {}, expected_params)
+        mock_client.activate()
+        self.handler.init_state_table(request_id, num_bundles)
```

Comment:
This test doesn't have an input/output contract. What are you testing? Can we assert anything after this function has executed?

Reply:
`init_state_table` eventually calls `put_item` on the Table resource. This test verifies the input params to `put_item`. OK to switch to `moto` to enable output verification